src/Dapr.Client/TypeConverters.cs

```diff
@@ -21,65 +21,36 @@ internal static class TypeConverters
         /// <param name="options">The JSON serialization options.</param>
         /// <typeparam name="T">The type of the given data.</typeparam>
         /// <returns>The given data as JSON based byte string.</returns>
-        public static ByteString ToJsonByteString<T>(T data, JsonSerializerOptions options = null)
+        public static ByteString ToJsonByteString<T>(T data, JsonSerializerOptions options)
         {
-            if (data == null)
-            {
-                return ByteString.Empty;
-            }
```

Comment:
This change because `null` is absolutely a meaningful value in JSON. Different parsers handle "empty data" differently and some of them will barf on "empty" which makes it a bad candidate for silent conversion.